encoding/dot/decode_test.go

```diff
@@ -61,7 +51,7 @@ func TestRoundTrip(t *testing.T) {
 	}
 }
 
-const directed = `digraph G {
+const directed = `digraph {
```

Comment:
> Does the `Unmarshal` call still work with the graph label present?

It still works, but the graph ID information is lost. We could implement an `UnmarshalDOTID` or something along those lines, so that the ID may be unmarshalled, and later marshalled by graphs implementing the `DOTID` method.

Note, there are several other pieces of information in DOT files that is currently lost, and I don't know if the graph/encoding/dot package should aim for full coverage; the API would end up looking rather horrible I reckon.

For more involved decoding needs, users may import formats/dot directly.

I am definitely open to suggestions, both on how the API may look like and where to set the bar. Perhaps it makes sense to include unmarshalling support for graph IDs in dot.Builder, but ignore information such as the strictness of a graph, or other very Graphviz DOT-specific details.

You may sway my mind, with convincing arguments though :)